lib/Conversion/TorchToLinalg/Linear.cpp

```diff
@@ -943,37 +943,123 @@ class ConvertAtenConvolutionOp : public OpConversionPattern<AtenConvolutionOp> {
     // - grouped 1d-3d (quantized)
     // - ungrouped 1d-3d
     if (groupSize == 1 && !inputZp && !weightZp) {
+      // Get default accumulator tensor type for the result tensor and use
+      // it for upcast/downcast if it doesn't match with the result type.
+      Value accumulatorTensor;
+      Type accumulatorTensorType = getDefaultAccType(rewriter, resultDTy);
+      if (accumulatorTensorType != resultDTy) {
```

Comment:
You should be handling this here:
https://github.com/llvm/torch-mlir/pull/3017/files#diff-b9279b005b7ec749040c06d16d460b492e48b69d5d833dc3021454725de59b84R881

You are overriding / throwing away the outputTensor rather than correcting the generated type at creation.

Reply:
The reason I did this way is because I didn't check/test if the accumulation logic remains for all other conv logics. I can change to cover other cases too